x-pack/plugins/security_solution/public/timelines/components/open_timeline/note_previews/index.tsx

```diff
@@ -74,6 +80,124 @@ const ToggleEventDetailsButtonComponent: React.FC<ToggleEventDetailsButtonProps>
 };
 
 const ToggleEventDetailsButton = React.memo(ToggleEventDetailsButtonComponent);
+
+const DeleteNoteConfirm = React.memo<{
+  closeModal: EuiConfirmModalProps['onCancel'];
+  confirmModal: EuiConfirmModalProps['onConfirm'];
+}>(({ closeModal, confirmModal }) => {
+  return (
+    <EuiConfirmModal
+      title={i18n.DELETE_NOTE_CONFIRM}
+      onCancel={closeModal}
+      onConfirm={confirmModal}
+      cancelButtonText={i18n.CANCEL_DELETE_NOTE}
+      confirmButtonText={i18n.DELETE_NOTE}
+      buttonColor="danger"
+      defaultFocusedButton="confirm"
+    />
+  );
+});
+
+DeleteNoteConfirm.displayName = 'DeleteNoteConfirm';
+
+function useDeleteNote(noteId: string | null | undefined) {
+  const {
+    services: { http },
+  } = useKibana();
+  const dispatch = useDispatch();
+  const { addError } = useAppToasts();
+  const onSuccess = useCallback(() => {
+    if (noteId) {
+      dispatch(
+        appActions.deleteNote({
+          id: noteId,
+        })
+      );
+    }
+  }, [dispatch, noteId]);
+
+  const onError = useCallback(
+    (err) => {
+      addError(err, { title: i18n.DELETE_NOTE_ERROR(err) });
+    },
+    [addError]
+  );
+
+  return useQuery(
```

Comment:
I think you want to use `useMutation` instead of `useQuery`?

Reply:
ya thank you, updated